security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/action/TapChangerRegulationActionDeserializer.java

```diff
@@ -0,0 +1,93 @@
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.powsybl.commons.json.JsonUtil;
+import com.powsybl.iidm.network.PhaseTapChanger;
+import com.powsybl.iidm.network.ThreeWindingsTransformer;
+import com.powsybl.security.action.*;
+
+import java.io.IOException;
+
+/**
+ * @author Etienne Lesot <etienne.lesot@rte-france.com>
+ * @author Anne Tilloy <anne.tilloy@rte-france.com>
+ */
+public class TapChangerRegulationActionDeserializer extends StdDeserializer<AbstractTapChangerRegulationAction> {
+
+    public TapChangerRegulationActionDeserializer() {
+        super(AbstractTapChangerRegulationAction.class);
+    }
+
+    private static class ParsingContext {
+        String id;
+        String type;
+        String transformerId;
+        boolean regulating;
+        ThreeWindingsTransformer.Side side = null;
+        String regulationMode;
+        Double regulationValue = null;
+        Double targetV = null;
+    }
+
+    @Override
+    public AbstractTapChangerRegulationAction deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+        ParsingContext context = new ParsingContext();
+        JsonUtil.parsePolymorphicObject(jsonParser, name -> {
+            switch (name) {
+                case "type":
+                    String type = jsonParser.nextTextValue();
+                    if (!PhaseTapChangerRegulationAction.NAME.equals(type) && !RatioTapChangerRegulationAction.NAME.equals(type)) {
+                        throw JsonMappingException.from(jsonParser, "Expected type " + PhaseTapChangerRegulationAction.NAME
+                                + "or " + PhaseTapChangerRegulationAction.NAME);
+                    }
+                    context.type = type;
+                    return true;
+                case "id":
+                    context.id = jsonParser.nextTextValue();
+                    return true;
+                case "transformerId":
+                    context.transformerId = jsonParser.nextTextValue();
+                    return true;
+                case "side":
+                    context.side = ThreeWindingsTransformer.Side.valueOf(jsonParser.nextTextValue());
+                    return true;
+                case "regulating":
+                    jsonParser.nextToken();
+                    context.regulating = jsonParser.getValueAsBoolean();
+                    return true;
+                case "regulationMode":
+                    context.regulationMode = jsonParser.nextTextValue();
+                    return true;
+                case "regulationValue":
+                    jsonParser.nextToken();
+                    context.regulationValue = jsonParser.getValueAsDouble();
+                    return true;
+                case "targetV":
```

Comment:
Here too it would be better to separate in 2 deserializers, possibly inheriting a common abstract class.

Otherwise, like on this line, we tolerate a field which is actually invalid for one of the target classes.